### PR TITLE
EAD-Session-1-Assignment

### PR DESCRIPTION
**Pull request by meta-jugal-kukreja**

### CODE REVIEW

File: EAD-Session-1-Assignment/functions.js [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+     * @param {*} password is the user password
+     * @param {*} confirmPassword is the confirm password field
+     * @param {*} contact is the user contact number
+     * @param {*} gender is the user gender
+     */
+    static validateUserInformation(name , email , password , confirmPassword , contact , gender)
+    {
+        // Validating Contact Number
+        var count=0;
+		var filterContact = new RegExp('([a-zA-Z!@#$%^&*()_+])');
```
Comment (meta-deepak-goyal) [inline EAD-Session-1-Assignment/functions.js]: Change the regex to allow only numeric characters. No need to use filterContact. Use the regex only for numeric allowed characters.

File: EAD-Session-1-Assignment/functions.js [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+		if(!(contact.length <= 10 && contact.length >= 8) ) {
+			this.errorDescription('required_e_contact',"Insert Number Between 8 to 10");
```
Comment (meta-deepak-goyal) [inline EAD-Session-1-Assignment/functions.js]: Change the error message like, contact number length is 8 to 10 character.

File: EAD-Session-1-Assignment/functions.js [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+			this.errorDescription('required_e_contact',"Numeric values are only allowed");
+		} else {
+			this.removeDescription('required_e_contact',"");
+			count--;
+		}
+
+        //Validating user name
+		var filterName = new RegExp('\\d');
+		if(name.length <= 2) {
+			this.errorDescription('required_e_name',"Insert name greater then 2 word");
```
Comment (meta-deepak-goyal) [inline EAD-Session-1-Assignment/functions.js]: Change message like username length can not be less than two characters.

File: EAD-Session-1-Assignment/functions.js [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+			this.removeDescription('required_e_email',"");
+			count--;
+		}
+
+        //Validating Password
+		var filterPassword = new RegExp("^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8,})");
+		if(password.length <= 8){
+			this.errorDescription('required_e_pass' ,"Atleast 8 char required");
+			count++;
+		} else if (!filterPassword.test(password)) {
```
Comment (meta-deepak-goyal) [inline EAD-Session-1-Assignment/functions.js]: Change message like password should have character, numeric values, special character, Upper case letter etc.

File: EAD-Session-1-Assignment/functions.js [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+			this.errorDescription('required_v_type',"Select Type Of Your Vehicle");
+			count++;
+		} else {
+			this.removeDescription('required_v_type',"");
+			count--;
+		}
+
+        //validating vehicle number
+		if(vehicleNumber.length < 4) {
+			this.errorDescription('required_v_number',"Vehicle Number required");
```
Comment (meta-deepak-goyal) [inline EAD-Session-1-Assignment/functions.js]: Change the error message like vehicle number length can not be less than 4 character. And also apply validation for vehicle number.

File: EAD-Session-1-Assignment/functions.js [hunk 6]
```diff
@@ -0,0 +1,373 @@
...
+     * Method to validate the vehicle information
+     * @param {*} vehicleName is the vehicle name
+     * @param {*} type is the vehicle type
+     * @param {*} vehicleNumber is the vehicle number
+     * @param {*} identity is the vehicle identity
+     */
+    static validateVehicleInformation(vehicleName , type , vehicleNumber , identity) {
+        var count = 0;
+        //validating vehicle name
+        if(vehicleName.length < 3)
```
Comment (meta-deepak-goyal) [inline EAD-Session-1-Assignment/functions.js]: Apply validation for vehicle name also.

File: EAD-Session-1-Assignment/functions.js (hunk 2 above)
Comment (meta-jugal-kukreja) [reply EAD-Session-1-Assignment/functions.js]: Done

File: EAD-Session-1-Assignment/functions.js (hunk 1 above)
Comment (meta-jugal-kukreja) [reply EAD-Session-1-Assignment/functions.js]: Done

File: EAD-Session-1-Assignment/functions.js (hunk 3 above)
Comment (meta-jugal-kukreja) [reply EAD-Session-1-Assignment/functions.js]: Done

File: EAD-Session-1-Assignment/functions.js (hunk 4 above)
Comment (meta-jugal-kukreja) [reply EAD-Session-1-Assignment/functions.js]: Done

File: EAD-Session-1-Assignment/functions.js (hunk 5 above)
Comment (meta-jugal-kukreja) [reply EAD-Session-1-Assignment/functions.js]: Done

File: EAD-Session-1-Assignment/functions.js (hunk 6 above)
Comment (meta-jugal-kukreja) [reply EAD-Session-1-Assignment/functions.js]: Done

**Pull request merged**